src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,320 @@
+    { menuText: 'Share', tooltipText: tooltipText.shareAzureUnsupported }
+  ])('renders $menuText menu item tooltip "$tooltipText"', ({ menuText, tooltipText }) => {
+    // Act
+    render(h(WorkspaceMenu, workspaceMenuProps))
+    fireEvent.mouseOver(screen.getByText(menuText))
+    // Assert
+    expect(screen.queryByText(tooltipText)).not.toBeNull()
+  })
+
+  it.each([
+    { menuText: 'Lock', accessLevel: 'OWNER' },
+    { menuText: 'Lock', accessLevel: 'READER' },
+    { menuText: 'Unlock', accessLevel: 'OWNER' },
+    { menuText: 'Unlock', accessLevel: 'READER' }
+  ])('enables/disables $menuText menu item based on access level $accessLevel', ({ menuText, accessLevel }) => {
+    // Arrange
+    useWorkspaceDetails.mockReturnValue({ workspace: { workspace: { isLocked: menuText === 'Unlock' }, accessLevel } })
+    // Act
+    render(h(WorkspaceMenu, workspaceMenuProps))
+    const menuItem = screen.getByText(menuText)
+    // Assert
+    if (Utils.isOwner(accessLevel)) {
+      expect(menuItem).not.toHaveAttribute('disabled')
+    } else {
+      expect(menuItem).toHaveAttribute('disabled')
+    }
+  })
+
+  it.each([
+    { menuText: 'Unlock', tooltipText: tooltipText.unlockNoPermission },
+    { menuText: 'Lock', tooltipText: tooltipText.lockNoPermission }
+  ])('renders $menuText menu item tooltip "$tooltipText" for access level READER', ({ menuText, tooltipText }) => {
+    // Arrange
+    useWorkspaceDetails.mockReturnValue({ workspace: { accessLevel: 'READER', workspace: { isLocked: menuText === 'Unlock' } } })
+    // Act
+    render(h(WorkspaceMenu, workspaceMenuProps))
+    fireEvent.mouseOver(screen.getByText(menuText))
+    // Assert
+    expect(screen.queryByText(tooltipText)).not.toBeNull()
+  })
+
+  it.each([
+    { accessLevel: 'READER', locked: true },
+    { accessLevel: 'OWNER', locked: true },
+    { accessLevel: 'READER', locked: false },
+    { accessLevel: 'OWNER', locked: false }
+  ])('renders Delete menu item as enabled/disabled for access level $accessLevel and locked status $locked', ({ accessLevel, locked }) => {
+    // Arrange
+    useWorkspaceDetails.mockReturnValue({ workspace: { workspace: { isLocked: locked }, accessLevel } })
+    // Act
+    render(h(WorkspaceMenu, workspaceMenuProps))
+    const menuItem = screen.getByText('Delete')
+    // Assert
+    if (!locked && Utils.isOwner(accessLevel)) {
+      expect(menuItem).not.toHaveAttribute('disabled')
+    } else {
+      expect(menuItem).toHaveAttribute('disabled')
+    }
+  })
+
+  it.each([
+    { accessLevel: 'READER', locked: true },
+    { accessLevel: 'OWNER', locked: true },
+    { accessLevel: 'READER', locked: false },
+    { accessLevel: 'OWNER', locked: false }
+  ])('renders Delete tooltip for access level $accessLevel and locked status $locked', ({ accessLevel, locked }) => {
+    // Arrange
+    useWorkspaceDetails.mockReturnValue({ workspace: { workspace: { isLocked: locked }, accessLevel } })
```

Comment:
I had a quick question: I see that `azureContext` is missing when mocking useWorkspaceDetails for some tests in azure case, should I include it?

Reply:
@yunip20 I pushed up the changes-- please review when you get a chance. Thanks!